src/server/main_service.cc

```diff
@@ -579,6 +579,24 @@ optional<Transaction::MultiMode> DeduceExecMode(ExecEvalState state,
   return multi_mode;
 }
 
+// Create a new owned argument list that includes the command name, only for debug assertions
+auto PrependCmdName(const CommandId* cid, CmdArgList args) {
+  struct Converge {
+    Converge(string_view s, CmdArgList tail) {
+      name_ = make_unique<string>(s);
+      args_ = {MutableSlice{name_->data(), name_->size()}};
+      args_.insert(args_.end(), tail.begin(), tail.end());
+    }
+    operator CmdArgList() {
+      return CmdArgList{args_};
+    }
+    unique_ptr<string> name_;
+    CmdArgVec args_;
+  };
+
+  return Converge{cid->name(), args};
+}
```

Comment:
Only purpose of this struct is to prolong the lifetime of the name & args vector beyond the function